models/user/rights.ts

```diff
@@ -31,6 +45,27 @@ export function hasRights(session: ISession | null, rightScope: EScope) {
   }
 }
 
+export function hasAnyRights(session: ISession | null, scopes: EScope[]) {
+  return scopes.some((scope) => hasRights(session, scope));
+}
+
 export function isLoggedIn(session: ISession | null) {
   return (session?.user?.scopes || []).length > 0;
 }
+
+export type INotAuthorized = {
+  __I_NOT_AUTHORIZED__: never;
+};
+
+export function isNotAuthorized<T>(
```

Comment:
We add a new base type for handling not authorized logic directly in the model when it makes sense (and not doing it systematically in the view).